pkg/nuclide-ocaml/lib/OutlineProvider.js

```diff
@@ -0,0 +1,90 @@
+'use babel';
+/* @flow */
+
+/*
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the LICENSE file in
+ * the root directory of this source tree.
+ */
+
+import type {Outline} from '../../nuclide-outline-view';
+import type {MerlinProcess} from '../../nuclide-ocaml-base/lib/MerlinProcess';
+
+import {Point} from 'atom';
+
+import {
+  className,
+  method,
+  keyword,
+  plain,
+  type,
+  whitespace,
+} from '../../commons-node/tokenizedText';
+import {getServiceByNuclideUri} from '../../nuclide-remote-connection';
+
+function makeTokens(data) {
+  let kind;
+  let nameToken;
+
+  switch (data.kind) {
+    case 'Value':
+      kind = 'val';
+      nameToken = method(data.name);
+      break;
+    case 'Class':
+    case 'Exn':
+    case 'Module':
+      nameToken = className(data.name);
+      break;
+    case 'Constructor':
+      kind = 'ctor';
+      nameToken = className(data.name);
+      break;
+    case 'Signature':
+      kind = 'sig';
+      nameToken = className(data.name);
+      break;
+    case 'Type':
+      nameToken = type(data.name);
+      break;
+  }
+  kind = kind || data.kind.toLowerCase();
```

Comment:
We prefer checking for null/undefined explicitly rather than relying on truthiness. For example if `kind` is `''` this will have confusing results. It's slightly more verbose but

```
if (kind != null) {
  kind = data.kind.toLowerCase();
}
```

captures the intention more exactly.